mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,714 @@
+        tl_heat = self.tl_heat[lvl_ind](tl_pool)
+        br_pool = self.br_pool[lvl_ind](x)
+        br_heat = self.br_heat[lvl_ind](br_pool)
+
+        tl_emb, br_emb = None, None
+        if self.with_corner_emb:
+            tl_emb = self.tl_emb[lvl_ind](tl_pool)
+            br_emb = self.br_emb[lvl_ind](br_pool)
+
+        tl_off = self.tl_off[lvl_ind](tl_pool)
+        br_off = self.br_off[lvl_ind](br_pool)
+
+        result_list = [tl_heat, br_heat, tl_emb, br_emb, tl_off, br_off]
+        if return_pool:
+            result_list.append(tl_pool)
+            result_list.append(br_pool)
+
+        return result_list
+
+    def corner_target(self,
+                      gt_bboxes,
+                      gt_labels,
+                      feat_shape,
+                      feat_dtype,
+                      feat_device,
+                      img_shape,
+                      with_corner_emb=False,
+                      with_guiding_shift=False,
+                      with_centripetal_shift=False):
+        b, _, h, w = feat_shape
+        img_h, img_w = img_shape[:2]
+
+        width_ratio = float(w / img_w)
+        height_ratio = float(h / img_h)
+
+        gt_tl_heatmap = torch.zeros([b, self.num_classes, h, w],
+                                    dtype=feat_dtype,
+                                    device=feat_device)
+        gt_br_heatmap = torch.zeros([b, self.num_classes, h, w],
+                                    dtype=feat_dtype,
+                                    device=feat_device)
+        gt_tl_offset = torch.zeros([b, 2, h, w],
+                                   dtype=feat_dtype,
+                                   device=feat_device)
+        gt_br_offset = torch.zeros([b, 2, h, w],
+                                   dtype=feat_dtype,
+                                   device=feat_device)
+
+        if with_corner_emb:
+            match = []
+
+        if with_guiding_shift:
+            gt_tl_guiding_shift = torch.zeros([b, 2, h, w],
+                                              dtype=feat_dtype,
+                                              device=feat_device)
+            gt_br_guiding_shift = torch.zeros([b, 2, h, w],
+                                              dtype=feat_dtype,
+                                              device=feat_device)
+        if with_centripetal_shift:
+            gt_tl_centripetal_shift = torch.zeros([b, 2, h, w],
+                                                  dtype=feat_dtype,
+                                                  device=feat_device)
+            gt_br_centripetal_shift = torch.zeros([b, 2, h, w],
+                                                  dtype=feat_dtype,
+                                                  device=feat_device)
+
+        for b_id in range(b):
+            corner_match = []
+            for box_id in range(len(gt_labels[b_id])):
+                tl_x, tl_y, br_x, br_y = gt_bboxes[b_id][box_id]
```

Comment:
If those abbreviations are invetible, at lease add some comments to help readers to understand.